pty/ptytest/ptytest.go

```diff
@@ -231,21 +231,21 @@ func (e *outExpecter) Peek(ctx context.Context, n int) []byte {
 	return slices.Clone(out)
 }
 
-func (e *outExpecter) ReadRune(ctx context.Context) rune {
+func (e *outExpecter) ReadRune() (rune, int, error) {
 	e.t.Helper()
 
 	var r rune
-	err := e.doMatchWithDeadline(ctx, "ReadRune", func(rd *bufio.Reader) error {
+	err := e.doMatchWithDeadline(context.Background(), "ReadRune", func(rd *bufio.Reader) error {
```

Comment:
This context will never have any deadline set, so this method could hang forever.

Reply:
I leave it here for @coder/eng to fix it :) Thank you.

Reply:
I guess we could revert the changes and rename the method. That said, vet is rarely used as enforcement because it came produce a lot of false positives.

The only "issue" here is that the method is named similar to an stdlib interface, but has a different signature. That's not enforcement worthy IMO because this clearly has a different goal than the interface does.

Reply:
TBH I didn't actually see this method being used, so why keep it around?

Reply:
It probably was used at some point, but I don’t have much preference if it’s kept or removed.

But I mainly I’m just worried about enforcing vet. I don’t think nolint works either if we’re using it outside golangci-lint like this.